src/snakemake/deployment/__init__.py

```
@@ -0,0 +1,109 @@
+                plugin_name
+                not in self.workflow.workflow_settings.experimental_sofware_deployment_methods
+            ):
+                continue
+            kind = plugin.common_settings.kind
+            if kind not in self.plugins:
+                self.plugins[kind] = plugin
+            else:
+                raise WorkflowError(
+                    "Multiple plugins for the same kind are not allowed. "
+                    f"Choose one of {plugin_name} and {self.plugins[kind].name}."
+                )
```

Ruff (0.14.6)

Avoid specifying long messages outside the exception class

(TRY003)